src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -182,6 +188,32 @@ private void applyStringFilter(String filterString) {
         }
     }
 
+    private void indicatePanelLoading(RepoOpeningEvent e) {
+        togglePanelLoading(e.isPrimaryRepo, e.repoId, true);
+    }
+
+    private void unindicatePanelLoading(RepoOpenedEvent e) {
+        togglePanelLoading(e.isPrimaryRepo, e.repoId, false);
+    }
```

Comment:
For the above two methods, a better structure is:

```java
private void indicatePanelLoading(RepoOpeningEvent e) {
       if(isIndicatorApplicable(e.isPrimaryRepo, e.repoId){
          showLoadingIndicator();
      }
 }
```

In the current code, you have abstracted the wrong action. It's checking the applicability that is common, not the toggling.